test/integration/scheduler/scheduler_test.go

```diff
@@ -199,20 +200,18 @@ var _ = ginkgo.Describe("Scheduler", func() {
 		}, framework.Timeout, framework.Interval).Should(gomega.Equal(pointer.Bool(false)))
 		gomega.Expect(createdJob1.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(onDemandFlavor.Name))
 
-		// TODO(#8): uncomment the following once we have proper re-queueing.
-		// ginkgo.By("checking a second job without toleration doesn't start")
-		// job2 := testing.MakeJob("on-demand-job2", namespace).Queue(queue.Name).Request(corev1.ResourceCPU, "5").Obj()
-		// gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
-		// createdJob2 := &batchv1.Job{}
-		// gomega.Consistently(func() bool {
-		// 	lookupKey := types.NamespacedName{Name: job2.Name, Namespace: job2.Namespace}
-		// 	return k8sClient.Get(ctx, lookupKey, createdJob2) == nil && *createdJob2.Spec.Suspend
-		// }, consistentDuration, interval).Should(gomega.BeTrue())
+		ginkgo.By("checking a second job without toleration doesn't start")
+		job2 := testing.MakeJob("on-demand-job2", ns.Name).Queue(prodBEQueue.Name).Request(corev1.ResourceCPU, "5").Obj()
+		gomega.Expect(k8sClient.Create(ctx, job2)).Should(gomega.Succeed())
+		createdJob2 := &batchv1.Job{}
```

Comment:
you can put this inside the `Consistently` clause since it is not used later.

Reply:
You could use QueuedWorkload (instead of Job) and use this https://github.com/kubernetes-sigs/kueue/blob/425187b36a282851e08ad25d0fd51fc0ee5fc2d9/test/integration/framework/framework.go#L151-L168

The advantage is that it uses Eventually instead of consistently, using the QW status.

Reply:
Can we please move forward with this PR? I think we can keep this test as is without changing to workload, we can do the migration for the whole file separately.

Reply:
Also change to use Workload (instead of Job) in this test case and use `ExpectWorkloadsToBePending` here.

> we can do the migration for the whole file separately.

+1 We can use workload in new added test case. And migration the old tests for the whole file in a separate pr.